src/typedarray.rs

```diff
@@ -0,0 +1,314 @@
+            }
+        }
+
+        impl TypedArrayElementCreator for $t {
+            unsafe fn create_new(cx: *mut JSContext, length: u32) -> *mut JSObject {
+                $create_new(cx, length)
+            }
+
+            unsafe fn get_data(obj: *mut JSObject) -> *mut Self::Element {
+                let mut shared = false;
+                let data = $get_data(obj, &mut shared, ptr::null_mut());
+                assert!(!shared);
+                data
+            }
+        }
+    );
+}
+
+typed_array_element!(Uint8,
+                     u8,
+                     UnwrapUint8Array,
+                     GetUint8ArrayLengthAndData,
+                     JS_NewUint8Array,
+                     JS_GetUint8ArrayData);
+typed_array_element!(Uint16,
+                     u16,
+                     UnwrapUint16Array,
+                     GetUint16ArrayLengthAndData,
+                     JS_NewUint16Array,
+                     JS_GetUint16ArrayData);
+typed_array_element!(Uint32,
+                     u32,
+                     UnwrapUint32Array,
+                     GetUint32ArrayLengthAndData,
+                     JS_NewUint32Array,
+                     JS_GetUint32ArrayData);
+typed_array_element!(Int8,
+                     i8,
+                     UnwrapInt8Array,
+                     GetInt8ArrayLengthAndData,
+                     JS_NewInt8Array,
+                     JS_GetInt8ArrayData);
+typed_array_element!(Int16,
+                     i16,
+                     UnwrapInt16Array,
+                     GetInt16ArrayLengthAndData,
+                     JS_NewInt16Array,
+                     JS_GetInt16ArrayData);
+typed_array_element!(Int32,
+                     i32,
+                     UnwrapInt32Array,
+                     GetInt32ArrayLengthAndData,
+                     JS_NewInt32Array,
+                     JS_GetInt32ArrayData);
+typed_array_element!(Float32,
+                     f32,
+                     UnwrapFloat32Array,
+                     GetFloat32ArrayLengthAndData,
+                     JS_NewFloat32Array,
+                     JS_GetFloat32ArrayData);
+typed_array_element!(Float64,
+                     f64,
+                     UnwrapFloat64Array,
+                     GetFloat64ArrayLengthAndData,
+                     JS_NewFloat64Array,
+                     JS_GetFloat64ArrayData);
+typed_array_element!(ClampedU8,
+                     u8,
+                     UnwrapUint8ClampedArray,
+                     GetUint8ClampedArrayLengthAndData,
+                     JS_NewUint8ClampedArray,
+                     JS_GetUint8ClampedArrayData);
+typed_array_element!(ArrayBufferU8,
+                     u8,
+                     UnwrapArrayBuffer,
+                     GetArrayBufferLengthAndData,
+                     JS_NewArrayBuffer,
+                     JS_GetArrayBufferData);
+
+
+/// The ArrayBufferView type.
+pub struct ArrayBufferViewU8;
+
+impl TypedArrayElement for ArrayBufferViewU8 {
```

Comment:
We shouldn't need this if we modify the typed_array_element macro as suggested above.